core/deployment/src/main/java/io/quarkus/deployment/dev/IsolatedDevModeMain.java

```diff
@@ -108,9 +109,28 @@ public void accept(Integer integer) {
                 runner = start.runMainClass(context.getArgs());
                 firstStartCompleted = true;
             } catch (Throwable t) {
-                deploymentProblem = t;
-                if (!augmentDone) {
-                    log.error("Failed to start quarkus", t);
+                Throwable rootCause = t;
+                while (rootCause.getCause() != null) {
+                    rootCause = rootCause.getCause();
+                }
+                if (rootCause instanceof BindException) {
+                    int port = ConfigProviderResolver.instance().getConfig()
+                            .getOptionalValue("quarkus.http.port", Integer.class).orElse(8080);
+                    log.error("Port " + port + " seems to be in use by another process. " +
+                            "Quarkus may already be running or the port is used by another application.");
+                    if (System.getProperty("os.name").startsWith("Windows")) {
+                        log.info("Use 'netstat -a -b -n -o' to identify the process occupying the port.");
+                        log.info("You can try to kill it with 'taskkill /PID <pid>' or via the Task Manager.");
+                    } else {
+                        log.info("Use 'netstat -anop | grep " + port + "' to identify the process occupying the port.");
+                        log.info("You can try to kill it with 'kill -9 <pid>'.");
+                    }
+                    System.exit(-1);
```

Comment:
What's the deal with this?

Reply:
And shouldn't we call `close();`?

Reply:
Good catch. I think `deploymentProblem = rootCause;` needs to be assigned here and `System.exit` shouldn't be explicitly called?

Reply:
Yeah, I don't see why `System.exit` is being called here. Can you explain?

Reply:
@gastaldi ok. I added it